qt/FreeSpaceLabel.cc

```diff
@@ -73,33 +66,35 @@ FreeSpaceLabel::onTimer ()
   tr_variantInitDict (&args, 1);
   tr_variantDictAddStr (&args, TR_KEY_path, myPath.toUtf8 ().constData());
 
-  myTag = mySession->getUniqueTag ();
-  mySession->exec ("free-space", &args, myTag);
-}
-
-void
-FreeSpaceLabel::onSessionExecuted (int64_t tag, const QString& result, tr_variant * arguments)
-{
-  Q_UNUSED (result);
+  RpcQueue * q = new RpcQueue;
 
-  if (tag != myTag)
-    return;
-
-  QString str;
-
-  // update the label
-  int64_t bytes = -1;
-  if (tr_variantDictFindInt (arguments, TR_KEY_size_bytes, &bytes) && bytes >= 0)
-    setText (tr("%1 free").arg(Formatter::sizeToString (bytes)));
-  else
-    setText (QString ());
-
-  // update the tooltip
-  size_t len = 0;
-  const char * path = 0;
-  tr_variantDictFindStr (arguments, TR_KEY_path, &path, &len);
-  str = QString::fromUtf8 (path, len);
-  setToolTip (QDir::toNativeSeparators (str));
+  q->addReq (
+    [this, &args] (RpcResponse)
```

Comment:
I was putting this off again and again, but I'm really bothered by the fact that you're passing most of the arguments in this patch by value. In most (if not all) cases, const reference is a better choice in my opinion. Particularly here (and in all the other Req/Proc lambdas), passing response by const reference is more correct conceptually, as we don't expect these callbacks to modify the response or move it some place else. Any solid ground behind this decision?